src/core/Templater.ts

```diff
@@ -124,7 +124,10 @@ export class Templater {
             folder,
             filename ?? "Untitled"
         );
-
+        
+        // Set activeFile to newly created note
+        await app.workspace.getLeaf(false).openFile(created_note, { state: { mode: "source" }, })
```

Comment:
This is happening on line 170 below if `open_new_note` is true. What is the purpose of adding it here?

Reply:
That allows calls in the template to use the currentFile if they do. Some plugins assert that they function on current file. I am using the Shortcut Launcher to edit some text with data I cannot get from obsidian directly and if I don't have this call it will affect the previous currentFile

Reply:
running the template is the part where I like having the currentfile set to what the template is.

Does that make sense? Daily notes using templater kind of does that

Reply:
Ok I understand, but should the logic that sets the current file be removed below? There is also a check in place if the file should be opened.

I want to make sure adding this won't break anyone's existing workflow.

Reply:
I guess the logic below could be partly removed. Let me review the code more.

For other workflows... Hard to tell